rct229/rulesets/ashrae9012019/section5/section5rule1.py

```diff
@@ -138,28 +138,28 @@ def get_calc_vals(self, context, data=None):
                     "$.output.output_instance", building_b270, False
                 )
 
-                # define a function to find the azimuth's corresponding key
-                find_key_for_azi = lambda azi: next(
-                    (
-                        key
-                        for key, range_ in azimuth_fen_area_dict_b.items()
-                        if int(key.split("-")[0]) <= azi <= int(key.split("-")[1])
-                    ),
-                    None,
-                )
-
-                azimuth_fen_area_dict_b = {
-                    f"{azi}-{azi+3}": ZERO.AREA for azi in range(3, 360, 3)
-                }
+                # define a function to get the azimuth's corresponding key
+                def get_key_for_azi(azi):
+                    azi_value = azi.to("degrees").magnitude
+                    low_bound = azi_value - (azi_value % 3)
+                    high_bound = low_bound + 3
+                    return f"{low_bound}-{high_bound}"
+
+                azimuth_fen_area_dict_b = {}
                 total_surface_fenestration_area_b = ZERO.AREA
                 for surface_b in find_all(
                     "$.building_segments[*].zones[*].surfaces[*]", building_b0
                 ):
                     if get_opaque_surface_type(surface_b) == OST.ABOVE_GRADE_WALL:
                         surface_azimuth_b = surface_b["azimuth"]
```

Comment:
Is the "azimuth" guaranteed? Likely to use a `getattr_` here.

Reply:
It is not listed as a required field but all above grade walls should have an azimuth.

Reply:
The azimuth key always has to exist and this was done with the ```required_fields``` attribute.